core/src/main/scala/org/apache/spark/rdd/CoGroupedRDD.scala

```diff
@@ -58,10 +58,20 @@ private[spark] case class NarrowCoGroupSplitDep(
  *                   narrowDeps should always be equal to the number of parents.
  */
 private[spark] class CoGroupPartition(
-    idx: Int, val narrowDeps: Array[Option[NarrowCoGroupSplitDep]])
+    override val index: Int, val narrowDeps: Array[Option[NarrowCoGroupSplitDep]])
   extends Partition with Serializable {
-  override val index: Int = idx
-  override def hashCode(): Int = idx
+
+  override def hashCode(): Int = index
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[CoGroupPartition]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: CoGroupPartition =>
+      super.equals(that) &&
```

Comment:
Same comment about Object equality here, and I don't think you need to check canEqual.